here-naksha-lib-base/src/jvmMain/kotlin/naksha/base/Platform.kt

```diff
@@ -32,6 +32,23 @@ import kotlin.reflect.full.primaryConstructor
 actual class Platform {
     actual companion object PlatformCompanion {
 
+        @JvmStatic
+        actual fun deepEquals(obj1: Any?, obj2: Any?): Boolean {
+            if (obj1 === obj2) return true  // Same reference, or both null
+            if (obj1 == null || obj2 == null) return false  // One is null, the other is not
+            if (obj1::class != obj2::class) return false  // Different types
+
+            return when (obj1) {
+                is Array<*> -> obj1.contentDeepEquals(obj2 as? Array<*>)
```

Comment:
Why do you cast via `as?`, you already tested that both, `obj1` and `obj2` are not _null_! (Same applies to all other casts below?)

Reply:
Will change to `as`, should be no difference as we already check that the types are the same. This `as?` just avoid throwing exceptions AFAIK, but on second thought maybe exception here would be better, as it is totally unexpected.

Reply:
<img width="747" alt="image" src="https://github.com/user-attachments/assets/675c1511-20f5-4339-a919-0c20986baf40">
I will try to look for this in official docs if needed

Reply:
<img width="742" alt="image" src="https://github.com/user-attachments/assets/9833257d-375a-4503-ad8a-6c3c6e930817">

Reply:
But you tested with `is Array<*>` before you cast via `as`, there is no way this should fail and there should be no way that the thing becomes _null_ in the meantime.